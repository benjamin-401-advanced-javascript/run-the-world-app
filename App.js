/* eslint-disable react-native/no-inline-styles */
import React, { useState } from 'react';
import {
  StyleSheet, Text, View, Button, TextInput,
} from 'react-native';

export default function App() {
  const [usernameText, setusernameText] = useState('');
  const [passwordText, setpasswordText] = useState('');

  const handleUsernameInput = (enteredText) => {
    setusernameText(enteredText);
  };

  const handleSigninSubmit = () => {
    // console.log(usernameText, passwordText);
  };

  return (
    <View style={styles.screen}>
      <View >

        <TextInput
          placeholder='Username'
          onChangeText={handleUsernameInput}
          value={usernameText}
        />
        <TextInput placeholder='Password' />
        <Button
          title='Sign In'
          onPress={handleSigninSubmit}
        />

      </View>
    </View>

  );
}

const styles = StyleSheet.create({
  screen: {
    padding: 40,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
