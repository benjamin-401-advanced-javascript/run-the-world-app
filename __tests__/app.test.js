// __tests__/app.test.js
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../src/app';

test('renders correctly', () => {
  const root = renderer.create(<App />);
  // const tree = renderer.create(<App />).toJSON();
  expect(root.toJSON()).toMatchSnapshot();
});
