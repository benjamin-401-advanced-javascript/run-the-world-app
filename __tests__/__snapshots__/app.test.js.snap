// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      placeholder="Password"
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
